### DISCUSSION TITLE
which bar/price does backtesting.py use to execute an order?

### DISCUSSION THREAD

**Discussion by ghostrider** [kernc/backtesting.py#432] [Q&A]
Hello all, 

in the sample code below, the SMACross strategy buys and sells an asset upon crossover. 
My question is, which price is used to do so? 

    class SmaCross(Strategy):
        # Define the two MA lags as *class variables*
        # for later optimization
        n1 = 10
        n2 = 20
        
        def init(self):
            # Precompute the two moving averages
            self.sma1 = self.I(SMA, self.data.Close, self.n1)
            self.sma2 = self.I(SMA, self.data.Close, self.n2)
        
        def next(self):
            #print(self.data)
            # If sma1 crosses above sma2, close any existing
            # short trades, and buy the asset
            if crossover(self.sma1, self.sma2):
                self.position.close()
                self.buy()

            # Else, if sma1 crosses below sma2, close any existing
            # long trades, and sell the asset
            elif crossover(self.sma2, self.sma1):
                self.position.close()
                self.sell()
    

A bar comes in (to the next method), the strategy evaluates that bar and then closes and buys. But when are the orders being executed? Using the close price of the incoming bar, or close price of the next bar, or open price of next bar? 

Does anyone know? 

Thanks
Ulrich

**Comment by kernc** [comment]
From the [quick start user guide](https://kernc.github.io/backtesting.py/doc/examples/Quick%20Start%20User%20Guide.html#Strategy):
> new orders are executed on the next candle's _open_ (or the current candle's _close_ if [`trade_on_close=True`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest))

**Reply by ghostrider** [reply]
thanks - it's hard to rtfm ;-)
